server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -196,30 +210,70 @@ public void onNewInfo(ClusterInfo info) {
         } else {
             listener.onResponse(null);
         }
+        // Get set of indices that are eligible to be automatically unblocked
+        // Only collect indices that are currently blocked
+        final String[] indices = state.routingTable().indicesRouting().keys().toArray(String.class);
+        Set<String> indicesToAutoRelease = Arrays.stream(indices)
+            .filter(index -> indicesToMarkIneligibleForAutoRelease.contains(index) == false)
+            .filter(index -> state.getBlocks().hasIndexBlock(index, IndexMetaData.INDEX_READ_ONLY_ALLOW_DELETE_BLOCK))
+            .collect(Collectors.toCollection(HashSet::new));
+
+        if (indicesToAutoRelease.isEmpty() == false) {
+            if (diskThresholdSettings.isAutoReleaseIndexEnabled()) {
+                logger.info("Releasing read-only allow delete block on indices: [{}]", indicesToAutoRelease);
```

Comment:
style nit:
```suggestion
                logger.info("releasing read-only-allow-delete block on indices [{}]", indicesToAutoRelease);
```

Also could you emit a `debug` message in the `else` branch saying that we're not releasing this block on these indices?